### ISSUE TITLE
Build error (mysql2 0.4.0)

### ISSUE THREAD

**Issue by Keeguon** [brianmario/mysql2#674]
Hey,

I saw you released mysql2 0.4.0. Unfortunately the build seems to be broken here's the following output I have when I try to install the gem:

```
Fetching: mysql2-0.4.0.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing mysql2:
        ERROR: Failed to build gem native extension.

    /opt/ruby-2.2.3/bin/ruby -r ./siteconf20150908-2469-wgpfoj.rb extconf.rb
checking for ruby/thread.h... yes
checking for rb_thread_call_without_gvl() in ruby/thread.h... yes
checking for rb_thread_blocking_region()... no
checking for rb_wait_for_single_fd()... yes
checking for rb_hash_dup()... yes
checking for rb_intern3()... yes
-----
Using mysql_config at /usr/bin/mysql_config
-----
checking for mysql.h... yes
checking for errmsg.h... yes
checking for mysqld_error.h... yes
-----
Setting libpath to /usr/lib/mysql
-----
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling mysql2_ext.c
compiling result.c
compiling statement.c
compiling infile.c
compiling client.c
cc1: warnings being treated as errors
client.c: In function ‘rb_connect’:
client.c:363: error: passing argument 3 of ‘mysql_options’ from incompatible pointer type
/usr/include/mysql/mysql.h:531: note: expected ‘const char *’ but argument is of type ‘time_t *’
client.c:370: error: passing argument 3 of ‘mysql_options’ from incompatible pointer type
/usr/include/mysql/mysql.h:531: note: expected ‘const char *’ but argument is of type ‘unsigned int *’
make: *** [client.o] Error 1

make failed, exit code 2

Gem files will remain installed in /opt/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/mysql2-0.4.0 for inspection.
Results logged to /opt/ruby-2.2.3/lib/ruby/gems/2.2.0/extensions/x86_64-linux/2.2.0-static/mysql2-0.4.0/gem_make.out
```

FYI here's my version of MySQL:

    mysql  Ver 14.14 Distrib 5.1.73, for debian-linux-gnu (x86_64) using readline 6.1

**Comment by sodabrew** [comment]
Oh yeah - they definitely changed the definition of that function!
https://dev.mysql.com/doc/refman/5.0/en/mysql-options.html
```
int mysql_options(MYSQL *mysql, enum mysql_option option, const char *arg)
```
vs.
https://dev.mysql.com/doc/refman/5.1/en/mysql-options.html
```
int mysql_options(MYSQL *mysql, enum mysql_option option, const void *arg)
```

I'm not sure why it's failing for you with MySQL 5.1, though - is it possible that you upgraded your MySQL packages but not the dev/header package?

**Comment by Keeguon** [comment]
@sodabrew that was it, my mistake I don't know why the dev/header package wasn't properly updated, my bad.

**Comment by sodabrew** [comment]
Oh cool, glad I could help! Actually I'm going to reopen the issue because because MySQL 5.0 is probably still in use enough to merit a workaround for it (although it's been EOL since 2012: http://dev.mysql.com/doc/relnotes/mysql/5.0/en/)

**Comment by sodabrew** [comment]
Resolved by #643
